### ISSUE TITLE
Context Menus are rendered incorrectly at high zoom levels

### ISSUE THREAD

**Issue by hoovercj** [microsoft/fluentui#16102]
### Environment Information

- **Package version(s)**: 7.X
- **Browser and OS versions**: Windows Edgeium

### Describe the issue:

When zooming the browser to 400%, context menus are rendered out of position and with a beak pointing to nothing. In addition to confusing usability, this could be considered a violation of MAS Reflow, MAS 1.4.10 Reflow ()

#### Actual behavior:

See screenshot:
![image](https://user-images.githubusercontent.com/2957173/100740850-a5f2ba80-33d9-11eb-829d-0e21dae466c4.png)

#### Expected behavior:

I would expect either the context menu shrinks further to fit in the space above/below/to the side of the anchor or, when not possible, simply renders as it does today **without a beak**, as the main issue is that that the beak is not pointing at the correct element.

### If applicable, please provide a codepen repro:

* Go to: https://fabricweb.z5.web.core.windows.net/oufr/7.153.2/#/examples/contextualmenu
* Find the example "ContextualMenu with beak and directional settings"
* Toggle the beak "on"
* Zoom to 400% on a screen that has a resolution of 1920x1080 or smaller (higher resolutions may also work but may require higher zoom
* Open the context menu

### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? **No**

Requested priority: **Normal**

Products/sites affected: **Dynamics 365 products (and many more)**

**Comment by JustSlone** [comment]
@hoovercj thanks for the feedback!

Yep, confirmed the repro. Actually I tweaked it a little to make the issue more obvious: [#16102 - Repro](https://codepen.io/justslone-the-encoder/pen/ZEpbPNL?editors=0010)

There is clearly code to add a scrollbar when the menu exceeds the screen size, however it's not handling small screens or high zoom levels. 

![image](https://user-images.githubusercontent.com/40304143/100965577-d6248100-34df-11eb-867a-705105b6cf91.png)

**Comment by smhigley** [comment]
Fixed in #19472
